software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+      set the visible of button "expandImgButt" stack "NodeDetails" to true 
+      -- the window is impossible to dismiss if these controls get covered up
+      set the layer of widget "iconOkay" to top
+      set the layer of widget "iconCancel" to top
+   end if -- the result is not cancel
+   set the cNodeDetails of stack "NodeDetails" to tDeets
+end loadImgButtMouseUp
+
+on clearImgButtMouseUp pLine
+   -- this button should only be available if there is something to clear
+   put the cNodeDetails of stack "NodeDetails" into tDeets
+   -- there can be a file in the graph, which will show up in [figure]
+   -- there can be a file just added in nodeDetails, which will show up in [new][copyfrom]
+   -- if a new file was just loaded, clear will get rid of the new one
+   -- if there's ONLY an existing file, clear will create a request to remove it from the graph
+   if tDeets[pLine]["new"] is not empty then
+      delete variable tDeets[pLine]["new"]
+      put false into tDeets["newFigure"]
+      if tDeets[pLine]["figure"] is "[empty]" then
+         -- cleared the just loaded image, so back to empty
+         set the filename of image "figureHolder" stack "NodeDetails" to empty
+         set the visible of button "clearImgButt" stack "NodeDetails" to false 
+         set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      else
+         -- cleared the just loaded image, so set the preview back to the image currently in the tag
+         put tDeets[pLine]["figure"] into tReq["targetMedia"]
+         put sMain[sNowDoc][0]["filepath"] into tReq["projectFile"]
+         set the filename of image "figureHolder" stack "NodeDetails" to correctAbsoluteMediaPath(tReq)
+         put the rect of field "editData" stack "NodeDetails" into tQuery["maxRect"]
+         put the long id of image "figureHolder" stack "NodeDetails" into tQuery["longID"]
+         put fitImageToConstraint(tQuery) into tGimme
+         set the width of image "figureHolder" stack "NodeDetails" to tGimme["width"]
+         set the height of image "figureHolder" stack "NodeDetails" to tGimme["height"]
+         set the loc of image "figureHolder" stack "NodeDetails" to tGimme["center"]  
+      end if -- [figure] is empty
+   else
+      -- [new] is empty, so remove existing tag from graph
+      put "hide" into tDeets[pLine]["new"]
+      set the filename of image "figureHolder" stack "NodeDetails" to empty
+   end if -- [new] is not empty
+   set the cNodeDetails of stack "NodeDetails" to tDeets
+end clearImgButtMouseUp
+
+on nodeDetailsChanged
+   -- technically, this means node details MIGHT have something new
+   put "initiate" into xX["write"];    devLog xX
+   put the cNodeDetails of stack "NodeDetails" into tDeets
+   -- save check; if there's a new figure, but this project's never been saved, then there's nowhere to put it
+   if tDeets["newFigure"] is true and sMain[sNowDoc][0]["filepath"] is empty then
+      saveAsButtMouseUp
```

Comment:
@matthewmaier Here's an example of why `ButtMouseUp` doesn't always make sense. See my previous comment.